server/src/test/java/io/crate/integrationtests/PostgresITest.java

```diff
@@ -1235,6 +1235,68 @@ public void test_fetch_with_intermediate_prepared_statement() throws Exception {
+    @Test
+    public void test_bulk_insert_from_select_fail_fast() throws Exception {
+        var properties = new Properties();
+        properties.setProperty("user", "crate");
+        properties.setProperty("options", "-c insert_fail_fast=true");
+        try (var conn = DriverManager.getConnection(url(RW), properties)) {
+            conn.createStatement().executeUpdate("create table t (a int primary key)");
+            ensureGreen();
+
+            PreparedStatement preparedStatement = conn.prepareStatement("insert into t (a) values (?)");
+            preparedStatement.setInt(1, 1);
+            preparedStatement.addBatch();
+            preparedStatement.setInt(1, 1);
+            preparedStatement.addBatch();
+
+            assertThatThrownBy(() -> preparedStatement.executeBatch())
+                .isExactlyInstanceOf(BatchUpdateException.class)
+                .hasMessageContaining("A document with the same primary key exists already");
+
+
+            // First error encountered is reflected in jobs_log.
+            var resultSet = conn.createStatement().executeQuery("""
+                SELECT error FROM sys.jobs_log WHERE stmt LIKE 'insert into t (a) values%'
+                """);
+            assertThat(resultSet.next()).isTrue();
+            assertThat(resultSet.getString(1)).contains("version conflict, document already exists");
+        }
+    }
+
+    @Test
+    public void test_temporal_to_be_deleted_checks_BulkShardResponseListener_code_path() throws Exception {
+        var properties = new Properties();
+        properties.setProperty("user", "crate");
+        properties.setProperty("options", "-c insert_fail_fast=true");
+
+        // BulkShardResponseListener is used in UpdateById + need some batches to force bulk execute
+
+        try (var conn = DriverManager.getConnection(url(RW), properties)) {
+            conn.createStatement().executeUpdate("create table t (a int CHECK (a < 100))");
+            conn.createStatement().executeUpdate("insert into t (a) values (1), (2), (3)");
+            conn.createStatement().executeUpdate("refresh table t");
+
+            PreparedStatement preparedStatement = conn.prepareStatement("update t set a = a + ?");
+            preparedStatement.setInt(1, 98);
+            preparedStatement.addBatch();
+            preparedStatement.setInt(1, 98);
+            preparedStatement.addBatch();
+
+            assertThatThrownBy(() -> preparedStatement.executeBatch())
+                .isExactlyInstanceOf(BatchUpdateException.class)
+                .hasMessageContaining("Failed CONSTRAINT");
+
+
+            // First error encountered is reflected in jobs_log.
+            var resultSet = conn.createStatement().executeQuery("""
+                SELECT error FROM sys.jobs_log WHERE stmt LIKE 'update t set a = a +%'
+                """);
+            assertThat(resultSet.next()).isTrue();
+            assertThat(resultSet.getString(1)).contains("Failed CONSTRAINT");
+        }
+    }
```

Comment:
I will remove this test. 

Basically, `BulkShardResponseListener` is already fine. I only updated it to reuse existing method, so no test is needed. 

Changes in `Session.emitRowCountsToResultReceivers` affect `BulkShardResponseListener` but for those changes I have a test.

Reply:
Hm, I pushed https://github.com/crate/crate/pull/15582/commits/9f8e8010a18a1ff434590a708be2a464d3dc94a2
which was actually doing `UpdateById` as described and apparently `BulkShardResponseListener` needs further adjustment. What I see is that single failure is null but failures contains cobstraint check failure.

I think need to iterate over failures, same as `ShardRequestExecutor` does.

Reply:
Looks like a bug on master - at least discrepancy between bulk/non-bulk update by id